src/to_wkt/geo_trait_impl.rs

```diff
@@ -0,0 +1,488 @@
+        geo_traits::Dimensions::Xyzm | geo_traits::Dimensions::Unknown(4) => {
+            f.write_str("LINESTRING ZM")
+        }
+        geo_traits::Dimensions::Unknown(_) => return Err(Error::UnknownDimension),
+    }?;
+    let size = PhysicalCoordinateDimension::from(dim);
+    if linestring.num_coords() == 0 {
+        Ok(f.write_str(" EMPTY")?)
+    } else {
+        add_coord_sequence(linestring.coords(), f, size)
+    }
+}
+
+pub fn write_polygon<T: WktNum + fmt::Display, G: PolygonTrait<T = T>, W: Write>(
+    polygon: &G,
+    f: &mut W,
+) -> Result<(), crate::error::Error> {
+    let dim = polygon.dim();
+    // Write prefix
+    match dim {
+        geo_traits::Dimensions::Xy | geo_traits::Dimensions::Unknown(2) => f.write_str("POLYGON"),
+        geo_traits::Dimensions::Xyz | geo_traits::Dimensions::Unknown(3) => {
+            f.write_str("POLYGON Z")
+        }
+        geo_traits::Dimensions::Xym => f.write_str("POLYGON M"),
+        geo_traits::Dimensions::Xyzm | geo_traits::Dimensions::Unknown(4) => {
+            f.write_str("POLYGON ZM")
+        }
+        geo_traits::Dimensions::Unknown(_) => return Err(Error::UnknownDimension),
+    }?;
+    let size = PhysicalCoordinateDimension::from(dim);
+    if let Some(exterior) = polygon.exterior() {
+        if exterior.num_coords() != 0 {
+            f.write_str("(")?;
+            add_coord_sequence(exterior.coords(), f, size)?;
+
+            for interior in polygon.interiors() {
+                f.write_char(',')?;
+                add_coord_sequence(interior.coords(), f, size)?;
+            }
+
+            Ok(f.write_char(')')?)
+        } else {
+            Ok(f.write_str(" EMPTY")?)
+        }
+    } else {
+        Ok(f.write_str(" EMPTY")?)
+    }
+}
+
+pub fn write_multi_point<T: WktNum + fmt::Display, G: MultiPointTrait<T = T>, W: Write>(
+    multipoint: &G,
+    f: &mut W,
+) -> Result<(), crate::error::Error> {
+    let dim = multipoint.dim();
+    // Write prefix
+    match dim {
+        geo_traits::Dimensions::Xy | geo_traits::Dimensions::Unknown(2) => {
+            f.write_str("MULTIPOINT")
+        }
+        geo_traits::Dimensions::Xyz | geo_traits::Dimensions::Unknown(3) => {
+            f.write_str("MULTIPOINT Z")
+        }
+        geo_traits::Dimensions::Xym => f.write_str("MULTIPOINT M"),
+        geo_traits::Dimensions::Xyzm | geo_traits::Dimensions::Unknown(4) => {
+            f.write_str("MULTIPOINT ZM")
+        }
+        geo_traits::Dimensions::Unknown(_) => return Err(Error::UnknownDimension),
+    }?;
+    let size = PhysicalCoordinateDimension::from(dim);
+
+    let mut points = multipoint.points();
+
+    // Note: This is largely copied from `add_coord_sequence`, because `multipoint.points()`
+    // yields a sequence of Point, not Coord.
+    if let Some(first_point) = points.next() {
+        f.write_str("((")?;
+
+        // Assume no empty points within this MultiPoint
+        add_coord(&first_point.coord().unwrap(), f, size)?;
```

Comment:
```suggestion
        write_coord(&first_point.coord().unwrap(), f, size)?;
```

Seems more in line with rest of the API, but given that it's private I don't care that much.